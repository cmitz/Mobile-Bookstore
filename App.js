import React from 'react';
import { SafeAreaView } from 'react-native';
import { Font } from 'expo';
import { createDrawerNavigator, createAppContainer } from 'react-navigation';
import { setCustomText } from 'react-native-global-props';
import { Provider } from 'mobx-react';
import { Ionicons } from '@expo/vector-icons';

import DashboardNavigator from './Components/Navigators/DashboardNavigator';
import StoresNavigator from './Components/Navigators/StoresNavigator';
import Sidebar from './Components/Sidebar';

import { RootStore } from './Stores/RootStore';

const AppNavigator = createDrawerNavigator({
  Dashboard: {
    screen: DashboardNavigator
  },
  Stores: {
    screen: StoresNavigator
  },
}, {
  initialRouteName: 'Dashboard',
  contentComponent: props => <Sidebar {...props} />,
  navigationOptions: {
    drawerLabel: 'Navigation Drawer',
  },
});


const AppContainer = createAppContainer(AppNavigator);

const rootStore = RootStore.create();

export default class App extends React.Component {
  state = {
    loading: true
  }

  async componentDidMount() {
    await Font.loadAsync({
      'Roboto': require('native-base/Fonts/Roboto.ttf'),
      'Roboto_medium': require('native-base/Fonts/Roboto_medium.ttf'),
      ...Ionicons.font,
    });
    setCustomText({ style: { fontFamily: 'Roboto' } });

    this.setState({ loading: false });
  }

  render() {
    const { loading } = this.state;
    return (
      <Provider rootStore={rootStore} >
        <SafeAreaView style={{ flex: 1, backgroundColor: '#ececec' }}>
          {loading ? <Expo.AppLoading />
            : <AppContainer />
          }
        </SafeAreaView>
      </Provider>
    );
  }
}
