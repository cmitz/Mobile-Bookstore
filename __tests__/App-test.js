import 'react-native';
import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { shallow } from 'enzyme';

import App from '../src/App';

jest.mock('react-dom');

describe('When App is rendered', () => {
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('does not crash', async () => {
    const tree = shallow(<App />);
    expect(tree).toBeTruthy();
  });

  it('renders the app like the snapshot', async () => {
    const tree = shallow(<App />);
    expect(tree).toMatchSnapshot();
  });
});

