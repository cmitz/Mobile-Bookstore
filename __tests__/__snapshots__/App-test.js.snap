// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When App is rendered renders the app like the snapshot 1`] = `<AppLoading />`;
